app/src/main/java/com/example/pokemonapp/viewmodel/SplashScreenViewModel.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.pokemonapp.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.pokemonapp.utils.Event
+import kotlinx.coroutines.Dispatchers.IO
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class SplashScreenViewModel : ViewModel() {
+
+    private var mutableSplashScreenState: MutableLiveData<Event<SplashScreenData>> = MutableLiveData()
+
+    fun getLiveDataState(): LiveData<Event<SplashScreenData>> = mutableSplashScreenState
+
+    fun navigateToMainFragment() = viewModelScope.launch {
+        mutableSplashScreenState.postValue(Event(SplashScreenData(responseType = SplashScreenStatus.LOADING)))
+        withContext(IO) {
+            delay(WAIT_TIME)
+            mutableSplashScreenState.postValue(Event(SplashScreenData(responseType = SplashScreenStatus.NAVIGATE, data = true)))
+        }
+    }
+
+    companion object {
+        private const val WAIT_TIME: Long = 4000
+    }
+}
+
+data class SplashScreenData(var responseType: SplashScreenStatus, var data: Boolean = false, var error: String = ERROR_DEFAULT) {
+    companion object {
+        private const val ERROR_DEFAULT: String = "error default"
```

Comment:
Please put this in the companion of the SplashScreenViewModel.
The data class and the enum class should be inside the SplashScreenViewModel too

Reply:
you have no service, so i cant imagine a error here. actually you never use the error, nor the data... so i think you should only use the SplashScreenStatus

Reply:
you re not usung neither the data nor the error. cause you dont call any service here.
i think you should delete this and only use the enum class

Reply:
@ezequielsetzesglobant Remember that we do not put things that we are not using now

Reply:
done